### ISSUE TITLE
Update ErrorProne to the latest version

### ISSUE THREAD

**Issue by TimvdLippe** [mockito/mockito#3493]
Over the past year (?) we have received Dependabot updates for ErrorProne. However, they fail the pipeline as the newest versions of ErrorProne catch existing issues in the codebase. An example PR from Dependabot: https://github.com/mockito/mockito/pull/3488

We need to update our version, as well as fix the pre-existing issues it catches. This is a good first issue to pick up, as it only requires compilation to succeed on the latest ErrorProne version.

**Comment by kbs1027** [comment]
Hi @TimvdLippe  , Can you please assign this issue to me??

**Comment by TimvdLippe** [comment]
@kbs1027 yes feel free to submit a PR. We don't assign issues to non-core developers, but we are happy to see your contribution!

**Comment by AmarReddy4** [comment]
I'll take a look at this. The previous PR #3783 was closed, so I'll pick it up from there — bumping to the latest ErrorProne (2.49.0), adding the JDK 21 javac flag it now requires, and updating the CI build JVM accordingly.
